set.mm

```diff
@@ -147154,6 +147148,405 @@ of an open range of nonnegative integers (of length equal to the length of
+     3-May-2020.) $)
+  pfxfv $p |- ( ( W e. Word V /\ L e. ( 0 ... ( # ` W ) )
+                    /\ I e. ( 0 ..^ L ) )
+                 -> ( ( W prefix L ) ` I ) = ( W ` I ) ) $=
+    ( cword wcel cc0 chash cfv cfz co cfzo w3a cpfx cop csubstr caddc wceq cn0
+    wi elfznn0 pfxval sylan2 3adant3 fveq1d cmin simp1 0elfz syl 3ad2ant2 simp2
+    nn0cnd subid1d eqcomd oveq2d eleq2d biimpd a1i 3imp swrdfv syl31anc addid1d
+    elfzoelz zcnd 3ad2ant3 fveq2d 3eqtrd ) DCEZFZBGDHIZJKFZAGBLKZFZMZADBNKZIADG
+    BOPKZIZAGQKZDIZADIVNAVOVPVIVKVOVPRZVMVKVIBSFZVTBVJUAZDBVHUBUCUDUEVNVIGGBJKF
+    ZVKAGBGUFKZLKZFZVQVSRVIVKVMUGVKVIWCVMVKWAWCWBBUHUIUJVIVKVMUKVIVKVMWFVKVMWFT
+    TVIVKVMWFVKVLWEAVKBWDGLVKWDBVKBVKBWBULUMUNUOUPUQURUSCDGBAUTVAVNVRADVMVIVRAR
+    VKVMAVMAAGBVCVDVBVEVFVG $.
+
+  $( Length of a prefix.  (Contributed by Stefan O'Rear, 24-Aug-2015.)
+     (Revised by AV, 2-May-2020.) $)
+  pfxlen $p |- ( ( S e. Word A /\ L e. ( 0 ... ( # ` S ) ) )
+                    -> ( # ` ( S prefix L ) ) = L ) $=
+    ( cword wcel cc0 chash cfv cfz co wa cpfx cfzo wfn pfxfn hashfn syl elfznn0
+    wceq cn0 adantl hashfzo0 eqtrd ) BADEZCFBGHZIJEZKZBCLJZGHZFCMJZGHZCUGUHUJNU
+    IUKSBCAOUJUHPQUGCTEZUKCSUFULUDCUERUACUBQUC $.
+
+  ${
+    $d x A $.  $d x S $.
+    $( A word is a prefix of itself.  (Contributed by Stefan O'Rear,
+       16-Aug-2015.)  (Revised by AV, 2-May-2020.) $)
+    pfxid $p |- ( S e. Word A -> ( S prefix ( # ` S ) ) = S ) $=
+      ( vx cword wcel cc0 chash cfv cfzo co cpfx cfz wf lencl nn0fz0 sylib pfxf
+      cn0 mpdan ffnd wrdfn cv wa wceq simpl adantr simpr pfxfv syl3anc eqfnfvd
+      ) BADEZCFBGHZIJZBULKJZBUKUMAUNUKULFULLJEZUMAUNMUKULREUOABNULOPZULABQSTABU
+      AUKCUBZUMEZUCUKUOURUQUNHUQBHUDUKURUEUKUOURUPUFUKURUGUQULABUHUIUJ $.
+  $}
+
+  $( The range of a prefix of a word is a subset of the set of symbols for the
+     word.  (Contributed by AV, 2-May-2020.) $)
+  pfxrn $p |- ( ( W e. Word V /\ L e. ( 0 ... ( # ` W ) ) )
+                 -> ran ( W prefix L ) C_ V ) $=
+    ( cword wcel cc0 chash cfv cfz co wa cfzo cpfx pfxf frnd ) CBDEAFCGHIJEKFAL
+    JBCAMJABCNO $.
+
+  $( A prefix consisting of at least one symbol is not empty.  (Contributed by
+     Alexander van der Vekens, 4-Aug-2018.)  (Revised by AV, 2-May-2020.) $)
+  pfxn0 $p |- ( ( W e. Word V /\ L e. NN /\ L <_ ( # ` W ) )
+                 -> ( W prefix L ) =/= (/) ) $=
+    ( cword wcel cn chash cfv cle wbr w3a cpfx co c0 wne cc0 cfzo 3ad2ant2 wceq
+    cn0 lbfzo0 ne0i sylbir wf wb cfz simp1 nnnn0 lencl 3ad2ant1 simp3 syl3anbrc
+    elfz2nn0 pfxf syl2anc f0dom0 bicomd syl necon3bid mpbird ) CBDEZAFEZACGHZIJ
+    ZKZCALMZNOPAQMZNOZVBVAVHVDVBPVGEVHAUAVGPUBUCRVEVFNVGNVEVGBVFUDZVFNSZVGNSZUE
+    VEVAAPVCUFMEZVIVAVBVDUGVEATEZVCTEZVDVLVBVAVMVDAUHRVAVBVNVDBCUIUJVAVBVDUKAVC
+    UMULABCUNUOVIVKVJVFVGBUPUQURUSUT $.
+
+  $( The value of the prefix extractor is the empty set (undefined) if the
+     argument is not within the range of the word.  (Contributed by AV,
```

Comment:
Maybe I am not using the correct words, but I'd prefer something like:
> The value of a prefix operation when the integer argument is larger than the word length is the empty set. (This is due to our definition of function values for out-of-domain arguments, see ~iotanul ).

It was probably discussed already, but: have you thought about the definition which would give the whole word in that case (and more generally, similarly for substrings) ?

Reply:
Comment revised.

> It was probably discussed already, but: have you thought about the definition which would give the whole word in that case (and more generally, similarly for substrings) ?

No, I did not consider this, and I don't know if this was discussed by others. I cannot see any advantage of this at the moment, and the definition(s) would become wore complicated.